src/app/common/modals/calendar-modal/calendar-modal.component.html

```diff
@@ -0,0 +1,167 @@
+<div class="calendar-modal-component mat-typography">
+
+  <div fxLayout="row">
+    <div fxFlex="grow">
+      <h2 mat-dialog-title>Web calendar</h2>
+    </div>
+    <mat-slide-toggle
+      fxFlex
+      #webcalToggle
+      [matTooltip]="(webcal && webcal.enabled) ? 'Disable the web calendar' : 'Enable the web calendar'"
+      matTooltipPosition="above"
+      [checked]="webcal && webcal.enabled"
+      [disabled]="working"
+      (click)="onWebcalToggle()">
+    </mat-slide-toggle>
+  </div><!--/header-->
+
+  <div mat-dialog-content>
+
+    <ng-container *ngIf="!webcal">
+      <mat-progress-spinner
+        color="accent"
+        mode="indeterminate"
+        [diameter]="36"
+        style="margin: 0 auto">
+      </mat-progress-spinner>
+    </ng-container><!--/content-when-webcal loading-->
+
+    <ng-container *ngIf="webcal">
+
+      <small>Your web calendar is currently {{ webcal.enabled ? 'enabled' : 'disabled' }}.</small>
+
+      <div *ngIf="!webcal.enabled">
+        <p>
+          The web calendar displays due dates of your tasks as calendar events in an iCalendar that can be added to any
+          iCalendar client (e.g. Outlook/Google/iCloud calendar).
+        </p>
+      </div><!--/content-when-webcal-disabled-->
+
+      <div *ngIf="webcal.enabled">
+
+        <p>
+          The web calendar displays due dates of your tasks as calendar events. Use the following URL to subscribe to your
+          web calendar from your iCalendar client.
+        </p>
+        <div fxLayout="row">
+          <div fxFlex="grow" class="webcal-url-container">
+            <a [href]="bypass(webcalUrl)">{{webcalUrl}}</a>
```

Comment:
And this can be replaced with the SapePipe, with a small addition to it, so:

`[href]="webcalUrl | safe"...`